packages/batch/tests/unit/asyncProcessPartialResponse.test.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * Test processPartialResponse function
+ *
+ * @group unit/batch/function/processpartialresponse
+ */
+import type { Context, SQSEvent } from 'aws-lambda';
+import { helloworldContext as dummyContext } from '../../../commons/src/samples/resources/contexts';
+import { Custom as dummyEvent } from '../../../commons/src/samples/resources/events';
+import { AsyncBatchProcessor, asyncProcessPartialResponse } from '../../src';
+import { EventType } from '../../src/constants';
+import type { PartialItemFailureResponse } from '../../src/types';
+import { sqsRecordFactory } from '../helpers/factories';
+import { asyncSqsRecordHandler } from '../helpers/handlers';
+
+describe('Function: processPartialResponse()', () => {
+  const ENVIRONMENT_VARIABLES = process.env;
+  const context = dummyContext;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.resetModules();
+    process.env = { ...ENVIRONMENT_VARIABLES };
+  });
+
+  afterAll(() => {
+    process.env = ENVIRONMENT_VARIABLES;
+  });
+
+  describe('Process partial response function call tests', () => {
+    test('Process partial response function call with asynchronous handler', async () => {
+      // Prepare
+      const records = [
+        sqsRecordFactory('success'),
+        sqsRecordFactory('success'),
+      ];
+      const batch = { Records: records };
+      const processor = new AsyncBatchProcessor(EventType.SQS);
+
+      // Act
+      const ret = await asyncProcessPartialResponse(
+        batch,
+        asyncSqsRecordHandler,
+        processor
+      );
+
+      // Assess
+      expect(ret).toStrictEqual({ batchItemFailures: [] });
+    });
+
+    test('Process partial response through handler for SQS records with incorrect event type', async () => {
```

Comment:
The original tests were split into
- function call tests
- call through handler tests

Should we have the same split here? And have tests for each type of event to match the synchronous tests?